docs/options.md

```diff
@@ -327,6 +327,32 @@ Platform-specific variants also available:<br/>
 CIBW_TEST_EXTRAS: test,qt
 ```
 
+### `CIBW_BEFORE_TEST` {: #before-test}
+> Execute a shell command preparing test environment
+
+A shell command to run after setup test environment. This option allows you to run a command in **each** Python environment before the `pip wheel` command. This is useful if you need to install non pip package, change values of environment variables
+or perform multi step pip installation (like install `scikit-build` or `cython` before install test package)
```

Comment:
These reasons are a little curious, actually.

- Non-pip packages don't care about the virtualenv, so can be installed in BEFORE_BUILD. 
- Changing environment variables won't work, since the command is run in a subshell
- Maybe I'm missing something, but I can't see the need to install anything _before_ installing the wheel? Wheel installation is just an unzip and move, no code within the package is run. So why is there a need to install dependencies before that? It seems that TEST_REQUIRES would be sufficient for these cases.

Reply:
Example situation:

 - PyQt5 from 5.14.0 has wheel with `manylinux2014` platform tag. Earlier are published with `manylinux1`. `manylinux2014` needs pip  in version 19.3. Example solution with `BEFORE_BUILD` is to manually install `PyQt5==5.13.2`
- [cmake](https://pypi.org/project/cmake/) package does not provide wheels for x86 python 3.8. So only option is to build it from source. But cmake package needs `scikit-build` for build and this dependency needs to be importable (installed in previous transatction). And package [blosc](https://pypi.org/project/blosc/) does not provide wheel and need cmake package to be installed. And tests need `blosc` package. 
- i do not have given example but it is possible to have similar story with `Cython` 

Also non pip package can care about virtualenv (location of python executable). Here example: https://github.com/LLNL/zfp

And change environment variable I think about `ENV1=1 ENV2=7 command`